lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/params/LightGBMParams.scala

```diff
@@ -455,6 +455,13 @@ trait LightGBMParams extends Wrappable
   def getNumIterations: Int = $(numIterations)
   def setNumIterations(value: Int): this.type = set(numIterations, value)
 
+  val saveAllIterations = new BooleanParam(this, "numIterations",
+    "If true, all iterations are included in the output model string." +
+      "If false, only iterations up to the 'bestIteration' are saved.")
+  setDefault(saveAllIterations -> true)
```

Comment:
ok, so you would prefer just to do the simple change and always use the bestIteration as the parameter?  and hence no new param?  that's fine with me, just confirming.  This will change the existing behavior, but I'm assuming you consider it a bug then.

Reply:
Possibly, if that has the desired effect -- what I was thinking though was more that this should default to `false` rather than `true`, if we want the default behavior to reflect LightGBM.

Of course, by the logic of reflecting LIghtGBM, we should not have a parameter at all and should just prune the model in the same manner that they do>?

Reply:
Yeah, now that you've pointed out how they prune their model iterations if they hit the early stopping, I think we should too in the wrapper.  Otherwise, we are providing a non-built-in feature that might conflict with other behavior in the future. So I'd vote for just changing it to always use bestIteration.

I don't see the point in defaulting to an option that changes the current behavior.  The point of creating that new parameter was to maintain backward compat with no change on the user's part.  If we consider it a bug (of sorts), then no point in maintaining backward compat. Just fix it to make it consistent with how LightGBM works on its own.

I guess the only reason to keep that parameter and default to native LightGBM pruning would be if it's useful for people who want to see the extra iterations that they wouldn't otherwise see.  Is that useful in your opinion? (I have no opinion on that). If not, then I'd remove the complexity of the extra param.

Reply:
I can understand that point of view. I guess it is less obvious to me that the old "behavior" is something we want to maintain backwards compatibility with or not. So as I understand it, when a user engages in early stopping, the desired behavior is that it actually acts as if early stopping occurred, doesn't pretend that it doesn't -- except for this one little export function. To my way of thinking that's pretty clearly a bug, not a feature, but not my circus, not my monkeys, so if people more responsible for this wrapper feel differently, fine with me.